### ISSUE TITLE
Stimulus 3 with TypeScript

### ISSUE THREAD

**Issue by swaggermeister** [hotwired/stimulus#457]
Couldn't find installation instructions for Stimulus 3 with TypeScript. This got it working for me:

```typescript
import { Application } from "@hotwired/stimulus";
import { definitionsFromContext } from "@hotwired/stimulus-webpack-helpers";

declare global {
  interface Window {
    Stimulus: Application;
  }
}

window.Stimulus = Application.start();
const context = require.context("../controllers", true, /\.ts$/);
window.Stimulus.load(definitionsFromContext(context));
```

**Comment by alexander-schranz** [comment]
I think it would be great if we have some documentation what is the best way to use stimilus with typescript. Specially also what is the correct way to define types for elements, targets, values. I'm currently failing using typescript in my controllers and set the `types` for them:

https://discuss.hotwired.dev/t/stimulus-and-typescript/2458/2

**Comment by marcoroth** [comment]
Resolved via #529 and #540
